unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ChannelMemberEntry.cs

```diff
@@ -0,0 +1,99 @@
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Chat.HUD
+{
+    public class ChannelMemberEntry : BaseComponentView, IComponentModelConfig<ChannelMemberEntryModel>
+    {
+        [SerializeField] internal TMP_Text nameLabel;
+        [SerializeField] internal ImageComponentView userThumbnail;
+        [SerializeField] internal GameObject onlineMark;
+        [SerializeField] internal GameObject offlineMark;
+        [SerializeField] internal Button optionsButton;
+        [SerializeField] internal RectTransform userContextMenuPositionReference;
+
+        [Header("Configuration")]
+        [SerializeField] internal ChannelMemberEntryModel model;
+
+        private UserContextMenu userContextMenu;
+
+        public ChannelMemberEntryModel Model => model;
+
+        public override void Awake()
+        {
+            base.Awake();
+
+            optionsButton.onClick.AddListener(() =>
+            {
+                userContextMenu.Show(model.userId);
+                Dock(userContextMenu);
+            });
+        }
+
+        public void Configure(ChannelMemberEntryModel newModel)
+        {
+            model = newModel;
+            RefreshControl();
+        }
+
+        public override void RefreshControl()
+        {
+            SetUserId(model.userId);
+            SetUserName(model.userName);
+            SetUserThumbnail(model.thumnailUrl);
+            SetUserOnlineStatus(model.isOnline);
+        }
+
+        internal void SetUserId(string userId)
```

Comment:
why all these methods are internal? if the purpose are the tests, you should have an only entry point to validate this through `Configure` method

Reply:
I don't understand well what you say about having "an only entry point to validate this". I want to create a unit test for each method in an independent way, and for this reason we declare them as `internal`. This is something that we're already doing in the rest of the unit tests for other features.

Why do you think it is not recommended to declare these methods as `internal`?

We don't always want to test something from the "source" that triggers the method (I think that wouldn't be an unit test) but directly execute the method that we want to test (as an unit test).

Reply:
we've talked about this, it is recommended to instead of making internal methods, call the `Configure` method with different parameters and then do the coverage on each case. Its about design and unit testing standards to check that the flow works through the public firm of the class to ensure that the expected entry point is working

Reply:
Done!